label_sync/main.go

```diff
@@ -353,22 +386,44 @@ func classifyLabels(labels []Label, required, archaic, dead map[string]Label, no
 	}
 }
 
-func syncLabels(config Configuration, repos RepoLabels) (RepoUpdates, error) {
+func copyLabelMap(originalMap map[string]Label) map[string]Label {
+	newMap := make(map[string]Label)
+	for k, v := range originalMap {
+		newMap[k] = v
+	}
+	return newMap
+}
+
+func syncLabels(config Configuration, org string, repos RepoLabels) (RepoUpdates, error) {
 	// Ensure the config is valid
 	if err := config.validate(); err != nil {
 		return nil, fmt.Errorf("invalid config: %v", err)
 	}
 
 	// Find required, dead and archaic labels
-	required := make(map[string]Label) // Must exist
-	archaic := make(map[string]Label)  // Migrate
-	dead := make(map[string]Label)     // Delete
-	classifyLabels(config.Labels, required, archaic, dead, time.Now(), nil)
+	defaultRequired := make(map[string]Label) // Must exist
+	defaultArchaic := make(map[string]Label)  // Migrate
+	defaultDead := make(map[string]Label)     // Delete
+	classifyLabels(config.Default, defaultRequired, defaultArchaic, defaultDead, time.Now(), nil)
 
 	var validationErrors []error
 	var actions []Update
 	// Process all repos
 	for repo, repoLabels := range repos {
+		var required, archaic, dead map[string]Label
+		// Check if we have more labels for repo
+		if labels, ok := config.Repos[org+"/"+repo]; ok {
+			// Duplicate required, dead and archaic labels to avoid modifying defaults
+			required = copyLabelMap(defaultRequired)
+			archaic = copyLabelMap(defaultArchaic)
+			dead = copyLabelMap(defaultDead)
+			classifyLabels(labels, required, archaic, dead, time.Now(), nil)
```

Comment:
Similarly here, let's clarify this API instead of working around the existing functionality